calculateContamination.wdl

```diff
@@ -51,46 +57,52 @@ workflow calculateContamination {
 #   Bam and index file(s) collected into a new array.
 # =======================================================
 
-    if ( inputType=="fastq" && defined(inputGroups) ){
-        Array[InputGroup] inputGroups_ = select_first([inputGroups])
-        scatter (ig in inputGroups_) {
+    if ( inputType=="fastq" && defined(fastqInputs) ){
+        Array[FastqInputs] fastqInputs_ = select_first([fastqInputs])
+        scatter (fq in fastqInputs_) {
             call bwaMem.bwaMem {
                 input:
-                    fastqR1 = ig.fastq1,
-                    fastqR2 = ig.fastq2,
-                    readGroups = ig.readGroups
+                    fastqR1 = fq.fastq1,
+                    fastqR2 = fq.fastq2,
+                    readGroups = fq.readGroups
             }
-            BamInputs indexedBamFiles = {
+            BamInputs alignedBamInputs= {
                 "bamFile":bwaMem.bwaMemBam,
-                "baiFile":bwaMem.bwaMemIndex
+                "baiFile":bwaMem.bwaMemIndex,
+                "sampleType": fq.sampleType
             }
         }
     }
 
 # =======================================================
-#   Check to see if bam files array has 1 or 2 files.
-#   Determines whether we run tumor/normal or tumor-only.
+#   Check to see if bam for nromal sample exists,
+#   to determine whether we run tumor/normal or tumor-only.
 # =======================================================
 
-    Array[BamInputs] bamFiles_ = select_first([bamFiles, indexedBamFiles])    
+    Array[BamInputs] bamInputs_ = select_first([bamInputs, alignedBamInputs])    
     
-    if ( length(bamFiles_)==1 ) {
+    if  (length(bamInputs_)==1 && bamInputs_[0].sampleType == "tumor"){
         call tumorOnlyMetrics {
             input:
-                tumorBamFile = bamFiles_[0].bamFile,
-                tumorBaiFile = bamFiles_[0].baiFile
+                tumorBamFile = bamInputs_[0].bamFile,
+                tumorBaiFile = bamInputs_[0].baiFile,
+                refVCF = refVCF,
+                modules =modules
         }
     }
-    if ( length(bamFiles_)==2 ) {
+    if  (length(bamInputs_)==2 && bamInputs_[0].sampleType == "tumor") {
```

Comment:
this should more fullly check
bamInputs_[0].sampleType == "tumor" && bamInputs_[1].sampleType == "normal"

Reply:
should we have an else here that will capture cases where not given either 1) tumour or 2) tumour + normal?   or will the workflow simply fail?

can we check on input earlier, so that if not given one of the two required configurations, it fails BEFORE alignment